pysumma/simulation.py

```diff
@@ -39,161 +39,165 @@ def __init__(self, executable, filemanager, run_suffix='_pysumma_run'):
-                env_vars = {'LOCALBASEDIR': self.base_dir,
-                            'MASTERPATH': self.filepath}
-                # define the location we want to mount these
-                # data in the container
-                vol_target = '/tmp/summa'
-                # define the base path of the input data for SUMMA
-                vol_source = self.base_dir
-                # run the container with the arguments specified above
-                res = jobs.run(specworker_img, '-x', vol_source,
-                               vol_target, env_vars)
+            out_file_path = (self.manager.output_path.value
+                             + self.manager.output_prefix.value + '_output_'
+                             + run_suffix + '_timestep.nc')
+        else:
+            self.executable = self.summa_code + '/bin/summa.exe'
+            cmd = "{} -p never -s {} -m {}".format(
+                self.executable, self.run_suffix, self.manager_path)
+            # run shell script in python and print output
+            cmd = shlex.split(cmd)
+            p = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+            output = p.communicate()[0].decode('utf-8')
+            print(output)
+            if 'FATAL ERROR' in output:
+                raise Exception("SUMMA failed to execute!")
+            # define output file name as sopron version of summa
+            out_file_path = (self.manager.output_path.value
+                             + self.manager.output_prefix.value + '_output_'
+                             + run_suffix + '_timestep.nc')
+        return out_file_path
 
-                list = glob.glob(self.base_dir
-                                 + self.output_path.value.split('>')[1]+"*.nc")
-                output_list = [x for x in list
-                               if self.output_prefix.value in x]
-                myoutput_list = ' '.join(output_list[1:])
-                if myoutput_list.count(self.output_prefix.value) > 1:
-                    output_name = (
-                        self.base_dir + self.output_path.value.split('>')[1]
-                        + self.output_prefix.value + '_'
-                        + self.decision_obj.simulStart.value[0:4] + '-'
-                        + self.decision_obj.simulFinsh.value[0:4] + '_'
-                        + self.run_suffix + '1.nc')
-                    merge_netcdf = ('ncrcat ' + myoutput_list
-                                    + ' -O ' + output_name)
-                    subprocess.run(merge_netcdf, shell=True)
-                    myList = ' '.join(output_list[:])
-                    delete_netcdf = 'rm -rf ' + myList
-                    subprocess.run(delete_netcdf, shell=True)
+    def exec_hydroshare(self, run_suffix, specworker_img=None):
+        #TODO: This needs to be updated
+        raise NotImplementedError('This needs to be updated')
+        self.run_suffix = ""
+        #TODO: This should be up top
+        from specworker import jobs
```

Comment:
This command works only in hydroshare execution. If I move this command at the top, the error occurs.